zkevm-circuits/src/evm_circuit/util/constraint_builder.rs

```diff
@@ -988,12 +988,12 @@ impl<'a, F: Field> ConstraintBuilder<'a, F> {
     // Stack
 
     pub(crate) fn stack_pop(&mut self, value: Expression<F>) {
-        self.stack_lookup(false.expr(), self.stack_pointer_offset.expr(), value);
-        self.stack_pointer_offset += 1;
+        self.stack_lookup(false.expr(), self.stack_pointer_offset.clone(), value);
+        self.stack_pointer_offset = self.stack_pointer_offset.clone() + self.condition_expr();
     }
 
     pub(crate) fn stack_push(&mut self, value: Expression<F>) {
-        self.stack_pointer_offset -= 1;
+        self.stack_pointer_offset = self.stack_pointer_offset.clone() - self.condition_expr();
```

Comment:
I am wandering why previous look up succeed:  the stack_pointer_offset is always reduced by 1 no matter the condition is.

Reply:
I think that is because it's only used when the condition is None.

Reply:
The reason that that works in the log gadget is because there are not additional stack lookups afterwards. The `StepStateTransition` in the gadget correctly (but inconsistent with `stack_pointer_offset`) specifies that the stack point only changes by `2 + topic_count`, so the error doesn't propagate into subsequent gadgets.